precompile/modules/initia_stdlib/sources/dex.move

```diff
@@ -981,14 +1004,118 @@ module initia_std::dex {
         coin::deposit(signer::address_of(account), return_coin);
     }
 
+    /// Flash swap is a special swap that allows the user to use return coin first and repay the offer coin later.
+    /// `data` is stargate message data that will be executed with swap return coin.
+    public entry fun flash_swap(
```

Comment:
not too sure if i follow, but it feels like we want to leverage [hot-potato](https://move-book.com/programmability/hot-potato-pattern.html) to do flashswap. an example can be found at [liquidswap flashloan](https://github.com/pontem-network/liquidswap/blob/main/sources/swap/liquidity_pool.move#L368) and you may find [this example](https://docs.liquidswap.com/liquidswap-v1-docs/integration/flashloans) helpful.

the idea is `FlashSwap` struct wont have any ability so that it must be constructed and destroyed in a same transaction. under the context of move, not sure if we still need `data: vector<u8>`.

lmk if any question

Reply:
Thanks for recommendation, I just updated to use liquidswap style.